MariaIzamara/web-comp-analyzer
# Feat: Initial setup with activity bar icon, docs.json integration, welcome view and tree-view

This pull request sets up the initial structure and functionality of the extension, which is designed to work with `docs.json` files commonly generated in Stencil projects. The extension's features ensure usability even if no `docs.json` is available or detected.

### Changes
1. Added an icon in the activity bar:
A custom icon provides quick access to the extension directly from the VSCode activity bar.
2. Integrated `docs.json` detection and reading:
The extension searches for a `docs.json` file in the currently opened folder. If found, it reads and uses the file as its primary data source.
3. Created a tree-view for components:
A left sidebar in the VSCode interface displays a hierarchical view of components, using data from the `docs.json` file.
4. Configured the welcome view:
The welcome view is displayed in the following scenarios:
    - No folder is opened.
    - The opened folder does not contain a `docs.json` file.
    - The `docs.json` file could not be found or accessed.
From the welcome view, users can either open a folder or directly select a docs.json file.
5. Implemented the `extension.openFile` command:
A command to open a specific file from the welcome view, allowing users to quickly access or generate a docs.json if needed.

### Testing Instructions
- Verify that the activity bar icon is visible and functional.
- Open folders with and without a `docs.json` file to confirm the welcome view behavior.
- Use the welcome view to open a folder or select a specific `docs.json` file.
- Confirm the tree-view loads and displays the component hierarchy when a valid `docs.json` file is present.
- Test the `extension.openFile` command to ensure it opens the intended file.
